java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/persistence/DynamoDbMetastoreImplTest.java

```diff
@@ -216,17 +218,54 @@ void testStoreWithFailureShouldThrowException() {
   void testPrimaryBuilderPath() {
     // Hack to inject default region since we don't explicitly require one be specified as we do in KMS impl
     System.setProperty(SDKGlobalConfiguration.AWS_REGION_SYSTEM_PROPERTY, "us-west-2");
+    DynamoDbMetastoreImpl dynamoDbMetastoreImpl = DynamoDbMetastoreImpl.newBuilder().build();
+
+    assertNotNull(dynamoDbMetastoreImpl);
+  }
+
+  @Test
+  void testBuilderPathWithRegion() {
+    DynamoDbMetastoreImpl dynamoDbMetastoreImpl = DynamoDbMetastoreImpl.newBuilder().withRegion("us-west-2").build();
 
-    DynamoDbMetastoreImpl dynamoDbMetastoreImpl =
-      DynamoDbMetastoreImpl.newBuilder().build();
     assertNotNull(dynamoDbMetastoreImpl);
   }
 
   @Test
   void testMetastoreWithRegionSuffix() {
-    DynamoDbMetastoreImpl dynamoDbMetastore = DynamoDbMetastoreImpl.newBuilder().withDynamoDbRegionSuffix("us-west-2").build();
+    DynamoDbMetastoreImpl dynamoDbMetastore = DynamoDbMetastoreImpl.newBuilder()
+      .withDynamoDbRegionSuffix("us-west-2")
+      .build();
 
     assertEquals("", dynamoDbMetastoreImpl.getRegionSuffix());
     assertEquals("us-west-2", dynamoDbMetastore.getRegionSuffix());
   }
+
+  @Test
+  void testMetastoreWithConfigurableTableName() {
+    String tableName = "DummyTable";
+
+    AmazonDynamoDB client = AmazonDynamoDBClientBuilder.standard()
+      .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("http://localhost:8000", "us-west-2"))
+      .build();
+    DynamoDB dynamoDBclient = new DynamoDB(client);
+
+    createTableSchema(dynamoDBclient, tableName);
+
+    Table table = dynamoDBclient.getTable(tableName);
+    Item item = new Item()
+      .withPrimaryKey(
+        PARTITION_KEY, TEST_KEY,
+        SORT_KEY, instant.getEpochSecond())
+      .withMap(ATTRIBUTE_KEY_RECORD, keyRecord);
+    table.putItem(item);
+
+    DynamoDbMetastoreImpl dynamoDbMetastore = DynamoDbMetastoreImpl.newBuilder()
+      .withEndPointConfiguration("http://localhost:8000", "us-west-2")
+      .withTableName(tableName)
+      .build();
+    Optional<JSONObject> actualJsonObject = dynamoDbMetastore.load(TEST_KEY, instant);
+
+    assertTrue(actualJsonObject.isPresent());
+    assertEquals(keyRecord, actualJsonObject.get().toMap());
+  }
```

Comment:
should we add one more test something like `testBuilderPathWithEndpointURL` because that is also a sufficient condition to build the metastore object and it'd be good to verify that it works correctly.

Reply:
The test itself run using the `withEndPoint` builder step so i think that covers that aspect

Reply:
All DynamoDb test setup is done using just the `withEndPointConfiguration` builder step. Do you still want to have an additional test?

Reply:
upon a second thought it seems un-necessary...but just in a future case if we have to modify the setup method in some scenario, then this test would ensure that the builder path works correctly.

Reply:
Adding to testing in isolation